### PR TITLE
Upgrade Towers

### LINKED ISSUE #36: Upgrading towers
You should be able to upgrade towers somehow (speed, radius, damage)

### PR DESCRIPTION
**Pull request by ross3102**
* Upgrade tower fire rate, damage, and radius
* Editable progression of values + costs

Screenshot 2024-03-29 at 11 42 13 PM

Screenshot 2024-03-29 at 4 14 31 PM

Closes #36

### CODE REVIEW

File: Assets/Scripts/LevelManager.cs [hunk 1]
```diff
@@ -123,4 +125,14 @@ public void EnemyDestroyed()
             LevelBeat();
         }
     }
+
+    public void SetPlaceTowerDisabled(bool disabled)
+    {
+        placeTowerDisabled = disabled;
+    }
+
+    public bool PlaceTowerDisabled()
```
Comment (meisbobzheng) [inline Assets/Scripts/LevelManager.cs:134]: IsPlaceTowerDisabled might be better naming convention

File: Assets/Scripts/ShootEnemies.cs [hunk 2]
```diff
@@ -17,18 +17,21 @@ public class ShootEnemies : MonoBehaviour
     Queue<GameObject> enemiesInRange;
     float lastShot;
     Transform cannonTransform;
+    CapsuleCollider collider;
 
     // Start is called before the first frame update
     void Start()
```
Comment (meisbobzheng) [inline Assets/Scripts/ShootEnemies.cs:23]: What is the purpose of this?

**Review by meisbobzheng — APPROVED** [review]

File: Assets/Scripts/ShootEnemies.cs (hunk 2 above)
Comment (ross3102) [reply Assets/Scripts/ShootEnemies.cs:23]: this is so we can resize the trigger to match the new radius after you upgrade the radius

Comment (meisbobzheng) [reply Assets/Scripts/ShootEnemies.cs:23]: ohhh i see

**Pull request merged**